Add Logging

BlakeASmith opened the pull request:
Resolves #66

igboyes reviewed (changes requested):
Inconsistent use of periods; don't use them.

The eventual goal is to connect all services to a logging service like [ELK](https://www.elastic.co/what-is/elk-stack). I think it is wise to only to log information at the `INFO` level that is important to system administrators. This would be stuff like filesystem problems, missing the database, and coarse workflow lifecycle. Implementation detail is important for debugging, but will clog up a logging server with a lot of useless information in production.

BlakeASmith commented on virtool_workflow_runtime/discovery.py:
```diff
@@ -87,18 +91,18 @@ def discover_workflow(path: Path) -> Workflow:
     workflow = next((attr for attr in module.__dict__.values() if isinstance(attr, Workflow)), None)
 
     if not workflow:
+        logger.info("No Workflow instance found, collecting startup, step, and cleanup functions.")
```
No, it could be put under DEBUG instead of INFO though. When the decorator API is used there is no instance of Workflow in the module, instead the functions tagged with @startup, @step, and @cleanup are collected in definition order and a new Workflow instance is made. So this would be a standard occurance.

BlakeASmith commented:
I've set the log levels appropriatley and removed any periods from the log messages.

I also added a new hook `on_load_config` and used it to set the log level to DEBUG when `dev_mode` is enabled. It is set to INFO otherwise.
